test/distributed/org/apache/cassandra/distributed/test/RepairErrorsTest.java

```diff
@@ -71,6 +81,52 @@ public void testRemoteValidationFailure() throws IOException
         }
     }
 
+    @SuppressWarnings("Convert2MethodRef")
+    @Test
+    public void testRemoteSyncFailure() throws Exception
+    {
+        try (Cluster cluster = init(Cluster.build(3)
+                                           .withConfig(config -> config.with(GOSSIP)
+                                                                       .with(NETWORK)
+                                                                       .set("disk_failure_policy", "stop")
+                                                                       .set("disk_access_mode", "mmap_index_only"))
+                                           .withInstanceInitializer(ByteBuddyHelper::installStreamPlanExecutionFailure).start()))
+        {
+            cluster.schemaChange("create table " + KEYSPACE + ".tbl (id int primary key, x int)");
+            
+            // On repair, this data layout will require two (local) syncs from node 1 and one remote sync from node 2:
+            cluster.get(1).executeInternal("insert into " + KEYSPACE + ".tbl (id, x) VALUES (?,?)", 1, 1);
+            cluster.get(2).executeInternal("insert into " + KEYSPACE + ".tbl (id, x) VALUES (?,?)", 2, 2);
+            cluster.get(3).executeInternal("insert into " + KEYSPACE + ".tbl (id, x) VALUES (?,?)", 3, 3);
+            cluster.forEach(i -> i.flush(KEYSPACE));
+            
+            // Flush system.peers_v2, or there won't be any SSTables...
+            cluster.forEach(i -> i.flush("system"));
+            
+            // The remote sync started from node 2 will fail on plan execution and propagate the error...
+            NodeToolResult result = cluster.get(1).nodetoolResult("repair", KEYSPACE);
+            result.asserts().failure().errorContains("Sync failed between /127.0.0.2:7012 and /127.0.0.3:7012");
+
+            assertTrue(cluster.get(1).logs().grep("Stopping transports as disk_failure_policy is stop").getResult().isEmpty());
```

Comment:
May want to leave a bread crumb comment here (referencing this JIRA + reasoning) as to why we're checking these three specific things to make sure they're not present in the logs.

Reply:
Yeah, I'll comment on how this actually failed before 17466. (i.e. I'll describe the steps to repro as well.)